### PR TITLE
Create a data access middleware

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
**Description**:

This pull request introduces a new feature: Data Access Middleware. This middleware allows for fine-grained control over which fields are exposed in API responses based on user roles and predefined rules.

**Changes Made**:

- Added a new middleware function, `dataAccessMiddleware`, which takes an `options` object as its configuration.
- The `options` object can include an array of `rules`, specifying which fields should be removed based on allowed user roles and key paths.
- The middleware iterates through the specified rules and removes sensitive fields from the response if the user does not have the required role.
- Added detailed JSDoc documentation for the `dataAccessMiddleware` function, including usage examples.

**Testing**:
Screenshot 2023-09-27 at 12 17 31 AM

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
test coverage is too low, 66 \% is really not acceptable

**Comment by Ajeyakrishna-k** [comment]
> test coverage is too low, 66 % is really not acceptable

@iamitprakash That was for other file which is not related to this PR. Ive updated the screenshot accordingly.

**Review by vinit717 — APPROVED** [review]

File: utils/data-access.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * app.use(dataAccessMiddleware({ rules }));
+ */
+const dataAccessMiddleware = (options = {}) => {
+  const rules = options.rules;
+
+  return (req, res, next) => {
+    try {
+      const { roles = {} } = req.userData;
+
+      const oldSend = res.send;
```
Comment (shubhamsigdar1) [inline utils/data-access.js:42]: Nit: What oldSend get?

can you help me to understand with this test case: https://github.com/Real-Dev-Squad/website-backend/pull/1564/files#diff-79f73954a237b17bfefc9e4db205ef4b93b4807163d96b0f2410cd599b1a5e27R34

**Review by shubhamsigdar1 — APPROVED** [review]
Great work @Ajeyakrishna-k, can you reply one of my comment

File: utils/data-access.js (hunk 1 above)
Comment (Ajeyakrishna-k) [reply utils/data-access.js:42]: If you're asking about where the assertion is taking place then please check this [link](https://github.com/Real-Dev-Squad/website-backend/pull/1564/files#diff-79f73954a237b17bfefc9e4db205ef4b93b4807163d96b0f2410cd599b1a5e27R10-R16).

Comment (Ajeyakrishna-k) [reply utils/data-access.js:42]: > Nit: What oldSend get?

send method in res object which will later be used inside custom send method that we are creating.

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**